_posts/2023-12-08-securing-wildfly-apps-auth0-openshift.adoc

```diff
@@ -0,0 +1,303 @@
+* https://auth0.com/[Auth0]
+
+== Example Application
+
+We will use a simple web application in this guide that consists of a single https://github.com/wildfly-security-incubator/elytron-examples/blob/main/simple-webapp-auth0/src/main/java/org/wildfly/security/examples/SecuredServlet.java[servlet]. We will secure this servlet using OIDC.
+
+We will use the example in the https://github.com/wildfly-security-incubator/elytron-examples/tree/main/simple-webapp-auth0[simple-webapp-auth0] directory in this repo.
+
+To obtain this example, clone the elytron-examples repository to your local machine:
+
+[source]
+----
+git clone git@github.com:wildfly-security-incubator/elytron-examples.git
+----
+
+== Log Into the OpenShift Cluster
+
+Before we can deploy our application, we need to log in to an OpenShift cluster. You can log in via the https://docs.openshift.com/container-platform/4.14/cli_reference/openshift_cli/getting-started-cli.html[OpenShift CLI]:
+
+[source]
+----
+oc login -u myUserName
+----
+
+Alternatively, you can log in using an API token:
+
+[source]
+----
+oc login --token=myToken --server=myServerUrl
+----
+
+You can request the token via the *Copy Login Command* link in the OpenShift web console.
+
+If you don't already have a project created, you can create one using:
+
+[source]
+----
+oc new-project myProjectName
+----
+
+== Configure Auth0
+
+We will be using Auth0 as our OpenID provider.
+
+. Log into the Auth0 Dashboard.
+
+. Create an application called *OIDC App*. For the application type, select *Regular Web Applications* and then click on *Create*. For more information, see the Auth0 documentation on how to https://auth0.com/docs/get-started/auth0-overview/create-applications[create applications].
+
+. Once the application has been created, we'll see the *Domain*, *Client ID*, and *Client Secret* in the *Basic Information* section. We'll make use of these values when https://wildfly-security.github.io/wildfly-elytron/blog/securing-wildfly-apps-auth0-openshift/#add-helm-configuration[adding Helm configuration] in a bit.
+
+. Using the sidebar menu on the left side of the Dashboard, navigate to the *APIs* page and copy the *API Audience* value.
+
+. Using the sidebar menu on the left side of the Dashboard, navigate to the *Settings* page and scroll down to the *API Authorization Settings*. Paste the API Audience value you just copied into the *Default Audience* field and then click on *Save*.
+
++
+This will allow us to receive access tokens that are JWTs from Auth0. In the future, we're hoping to add the ability
+to handle opaque access tokens as well to WildFly's Elytron OIDC Client subsystem.
+
+. Using the sidebar menu on the left side of the Dashboard, click on *User Management* and then *Users*. You can then
+create a new user by clicking on *Create User*. You'll need to specify the new user's email, we'll use *user@example.com*. You'll also need to set a password for the user.
```

Comment:
Just a minor question, I noticed in the video that the user@example.com was already existing, and was just wondering if it is created automatically or it was created before recording of the video? Nonetheless, this looks good to me!

Reply:
I created the user@example.com user following the steps here before recording the video (just to make the video a bit shorter).